File: Modules/HumanResource/EmployeeProfile/MasterCareerPath/MasterCareerPathForm.cs
using Serenity.ComponentModel;
using System;

namespace HRMSoftware.EmployeeProfile.Forms;

[FormScript("EmployeeProfile.MasterCareerPath")]
[BasedOnRow(typeof(MasterCareerPathRow), CheckNames = true)]
public class MasterCareerPathForm
{
    [HalfWidth]
    public int CareerPathType { get; set; }
    [HalfWidth]
    public string CareerPathCode { get; set; }
    public int CategoryId { get; set; }
    [TextAreaEditor(Rows = 3)]
    public string Description { get; set; }

}

File: Modules/ServerTypes/EmployeeProfile/MasterCareerPathForm.ts
import { MasterCareerPathType } from "./MasterCareerPathType";
import { EnumEditor, StringEditor, TextAreaEditor, PrefixedContext } from "@serenity-is/corelib";
import { Category } from "./Category";
import { initFormType } from "@serenity-is/corelib/q";

export interface MasterCareerPathForm {
    CareerPathType: EnumEditor;
    CareerPathCode: StringEditor;
    CategoryId: EnumEditor;
    Description: TextAreaEditor;
}

export class MasterCareerPathForm extends PrefixedContext {
    static formKey = 'EmployeeProfile.MasterCareerPath';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!MasterCareerPathForm.init)  {
            MasterCareerPathForm.init = true;

            var w0 = EnumEditor;
            var w1 = StringEditor;
            var w2 = TextAreaEditor;

            initFormType(MasterCareerPathForm, [
                'CareerPathType', w0,
                'CareerPathCode', w1,
                'CategoryId', w0,
                'Description', w2
            ]);
        }
    }
}

[MasterCareerPathType, Category]; // referenced types